src/main/java/com/google/devtools/build/lib/rules/objc/AppleSkylarkCommon.java

```diff
@@ -253,8 +251,12 @@ public StructImpl linkMultiArchBinary(
   }
 
   @Override
-  public DottedVersion dottedVersion(String version) {
-    return DottedVersion.fromString(version);
+  public DottedVersion dottedVersion(String version) throws EvalException {
+    try {
+      return DottedVersion.fromString(version);
+    } catch (IllegalArgumentException e) {
+      throw new EvalException(null, e.getMessage());
```

Comment:
We shouldn't catch any `IllegalArgumentException` here.

Can you change `DottedVersion.fromString` to throw something else, so that we catch a more precise exception?
(using an unchecked exception is probably simpler to avoid changing the call sites)

Reply:
You mean a custom unchecked exception? I can do that. Or, is this the change you wanted to do on import?

Reply:
Yes.
If you want to do it, that's perfect.

Reply:
nit: Could you make this extend a RuntimeException, not an Error? https://docs.oracle.com/javase/specs/jls/se7/html/jls-11.html#jls-11.1.1 says that "Error is the superclass of all the exceptions from which ordinary programs are not ordinarily expected to recover".